modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -871,6 +886,78 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
           i + interface_id_shift + _interface_boundary_id_shift) = _interface_boundary_names[i];
     }
   }
+  // sector id
+  if (getParam<bool>("use_sector_id"))
+  {
+    const auto extra_id_index = mesh0->add_elem_integer(getParam<std::string>("sector_id_name"));
+    std::vector<dof_id_type> elem_sector_ids(mesh0->n_elem(), 0);
+    unsigned int id = 1;
+    unsigned int offset = 0;
+    for (unsigned int is = 0; is < _num_sides; ++is)
+    {
+      unsigned int nelem_sector =
+          mesh0->n_elem() * _num_sectors_per_side[is] /
+          (accumulate(_num_sectors_per_side.begin(), _num_sectors_per_side.end(), 0));
+      std::fill(
+          elem_sector_ids.begin() + offset, elem_sector_ids.begin() + offset + nelem_sector, id);
+      offset += nelem_sector;
+      ++id;
+    }
+    unsigned int i = 0;
+    for (auto & elem : mesh0->element_ptr_range())
+      elem->set_extra_integer(extra_id_index, elem_sector_ids[i++]);
+  }
+
+  if (getParam<bool>("use_ring_id"))
+  {
+    const auto extra_id_index = mesh0->add_elem_integer(getParam<std::string>("ring_id_name"));
+    std::vector<dof_id_type> elem_ring_ids(mesh0->n_elem(), 0);
+    unsigned int offset_side = 0;
+    for (unsigned int is = 0; is < _num_sides; ++is)
+    {
+      unsigned int nelem =
+          mesh0->n_elem() * _num_sectors_per_side[is] /
+          (accumulate(_num_sectors_per_side.begin(), _num_sectors_per_side.end(), 0));
+      unsigned int offset_ring = 0;
+      if (getParam<MooseEnum>("ring_id_assign_type") == "block_wise")
+      {
+        for (unsigned int ir = 0; ir < _ring_radii.size(); ++ir)
+        {
+          unsigned int nelem_annular_ring = _num_sectors_per_side[is] * _ring_intervals[ir];
+          if (_quad_center_elements && ir == 0)
+            nelem_annular_ring = _num_sectors_per_side[is] * (_ring_intervals[ir] - 1) +
+                                 _num_sectors_per_side[is] * _num_sectors_per_side[is] / 4;
+          std::fill(elem_ring_ids.begin() + offset_side + offset_ring,
+                    elem_ring_ids.begin() + offset_side + offset_ring + nelem_annular_ring,
+                    ir + 1);
+          offset_ring += nelem_annular_ring;
+        }
+      }
+      else
+      {
+        unsigned int ir = 0;
+        for (unsigned int ir0 = 0; ir0 < _ring_radii.size(); ++ir0)
+        {
+          for (unsigned int ir1 = 0; ir1 < _ring_intervals[ir0]; ++ir1)
+          {
+            unsigned int nelem_annular_ring = _num_sectors_per_side[is];
+            if (_quad_center_elements && ir == 0)
+              nelem_annular_ring = _num_sectors_per_side[is] * _num_sectors_per_side[is] / 4;
+            std::fill(elem_ring_ids.begin() + offset_side + offset_ring,
+                      elem_ring_ids.begin() + offset_side + offset_ring + nelem_annular_ring,
+                      ir + 1);
+            ++ir;
```

Comment:
so in this case you want it not per ring, but per ring interval?

Reply:
Yes. if `ring_id_assign_type = ring_wise`, the ring ID are assigned to each ring interval.